### DISCUSSION TITLE
Naga: SPV to WGSL

### DISCUSSION THREAD

**Discussion by fhoenig** [gfx-rs/wgpu#5440] [Q&A]
My engine is written in Vulkan 1.2 with lots of GLSL fragment, vertex and compute programs. 
I just did a quick test with naga to see whether they can compile to WGSL.

From GLSL input some errors immediately don't seem reassuring:

Example:

```
error: Not implemented: variable qualifier
  ┌─ x.frag.glsl:7:38
  │
7 │ layout(set = 0, binding = 0) uniform sampler2D in_color;
  │                                      ^^^^^^^^^

```

and from SPV all of them error with:

```
naga TAA.vert.spv 
unsupported instruction Line at Type
```
The spirv is compiled with glslc (Google).

What would be the most robust path today for going from GLSL to WGSL (compiler versions, GLSL version, SPIR-V version, compiler arguments etc) ?

Thank you!

**Comment by teoxoy** [comment]
Naga doesn't currently support combined image/samplers (https://github.com/gfx-rs/wgpu/issues/4342, https://github.com/gfx-rs/wgpu/issues/4323) which seems like a hard blocker for your shaders.
